spec/classes/snmp_client_spec.rb

```diff
@@ -31,11 +30,7 @@
         let(:params) { {} }
         let :facts do
           {
-            osfamily: 'RedHat',
-            operatingsystem: os,
-            operatingsystemrelease: '6.4',
-            lsbmajdistrelease: '6',
-            operatingsystemmajrelease: '6'
+            os: { 'release' => { 'full' => '6.4', 'major' => '6' }, 'name' => 'CentOS', 'family' => 'RedHat' }
```

Comment:
How easy would it be to start using rspec-puppet-facts?

Reply:
Sure that tests with `rspec-puppet-facts` is better. But, before `4.0.0` is released i wished the module is with structured facts in its code.
If i have enough time before release, i will propose an other one about tests using `rspec-puppet-facts`. It is necessary to clean up some parts. I saw code/tests about old OSes.

For the moment, i propose to merge the PR as this.

Reply:
As long as the facts you've mocked are accurate and you've not just written them in such a way to make the tests pass! :)

Reply:
Excepted if i made an error, i reported same data but with structured facts instead of string facts

Reply:
Test with `rspec-puppet-facts` will be done in a futur PR.